src/shogun/machine/IterativeLinearMachine.cpp

```diff
@@ -0,0 +1,49 @@
+/*
+* This software is distributed under BSD 3-clause license (see LICENSE file).
+*
+* Authors: Shubham Shukla
+*/
+
+
+#include <shogun/lib/config.h>
+
+#include <shogun/lib/common.h>
+#include <shogun/machine/IterativeLinearMachine.h>
+
+using namespace shogun;
+
+CIterativeLinearMachine::CIterativeLinearMachine() : CLinearMachine()
+{
+ m_current_iteration = 0;
+ m_complete = false;
+ register_params();
+}
+
+CIterativeLinearMachine::~CIterativeLinearMachine()
+{
+}
+
+bool CIterativeLinearMachine::train(CDotFeatures* f, CLabels* l)
+{
+   init_model(f, l);
+   continue_train(f, l);
+   return m_complete;
+}
+
+void CIterativeLinearMachine::continue_train(CDotFeatures* f, CLabels* l)
+{
+  //resume_computation();
```

Comment:
this will need to be a method to revive computation controllers after cancel flag

Reply:
after fixing this a test like [here](https://gist.github.com/shubham808/c1446f80c0aca3bcfe695c507efed803) should work.

Reply:
See my comment below :)